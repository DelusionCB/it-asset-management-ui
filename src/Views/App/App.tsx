// Assets
import 'bootstrap/dist/css/bootstrap.min.css';
import '../../Assets/colors.scss';
import '../../Assets/fonts.scss'
import './index.scss'

import React from 'react';
import {Routes, Route, Outlet} from 'react-router-dom';

// Top level components
import Header from '../../Components/Header/Header';
import Footer from '../../Components/Footer/Footer';

// Views
import Homepage from '../Homepage/Homepage';
import ArchiveViewer from '../ArchiveViewer/ArchiveViewer';
import Search from '../Search/Search';
import ArchiveForms from '../ArchiveForms/ArchiveForms';

const App: React.FC = () => {
    return (
        <Routes>
            <Route path="/" element={<Layout />}>
                <Route index element={<Homepage />} />
                <Route path="/:path/:id" element={<ArchiveViewer />} />
                <Route path="/archive/:action/:id" element={<ArchiveForms />} />
                <Route path="/search" element={<Search />} />
            </Route>
        </Routes>
    );
}

const Layout: React.FC = () => {
    return (
        <div className='container-wrapper'>
            <Header />
            <main id='main'>
                <Outlet />
            </main>
            <Footer />
        </div>
    );
}

export default App;
