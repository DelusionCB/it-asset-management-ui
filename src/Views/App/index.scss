body {
    margin: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.container-wrapper {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
}

main {
    padding: 20px;
    flex: 1;
}
