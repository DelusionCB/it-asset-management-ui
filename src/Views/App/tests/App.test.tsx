import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import App from '../App';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import DirectoryProvider from '../../../Context/DirectoryContext';

describe('<App />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <DirectoryProvider>
                    <App />
                </DirectoryProvider>
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
