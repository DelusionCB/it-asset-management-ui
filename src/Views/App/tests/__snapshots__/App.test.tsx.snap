// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshot renders correctly 1`] = `
<div
  className="container-wrapper"
>
  <nav
    className="navbar navbar-expand navbar-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Varhan logo"
          height={80}
          src=""
        />
      </a>
      <div
        className="collapse navbar-collapse"
        style={{}}
      >
        <ul
          className="me-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <button
              aria-label="create-new"
              className="navlink-button btn btn-secondary"
              onClick={[Function]}
              role="link"
              type="button"
            >
              <span>
                create-new
              </span>
              <i
                className="bi bi-plus"
              />
            </button>
          </li>
          <li
            className="nav-item"
          >
            <button
              aria-label="archive-fetch"
              className="navlink-button btn btn-secondary"
              onClick={[Function]}
              role="link"
              type="button"
            >
              <span>
                archive-fetch
              </span>
              <i
                className="bi bi-search"
              />
            </button>
          </li>
        </ul>
      </div>
      <button
        aria-label="Toggle navigation"
        className="me-2 navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
    </div>
  </nav>
  <main
    id="main"
  >
    <div>
      <div>
        <h1>
          Tervetuloa IT- ja omaisuuden hallintaan!
        </h1>
      </div>
      <div>
        <div>
          <div
            className="directory-container"
          >
            <div>
              <h1>
                Ei kansioita joita näyttää
              </h1>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="footer"
  >
    <img
      alt="Varhan logo"
      height={80}
      src=""
    />
  </footer>
</div>
`;
